From e37dd635783d57e8e84bb32cd2abe701c9d705c3 Mon Sep 17 00:00:00 2001
From: Yongsheng Chen <yongsheng.chen@intel.com>
Date: Tue, 7 Aug 2018 15:15:22 -0700
Subject: [PATCH 2246/2328] media: intel-ipu4: ar0231: Fixed AR0231 overexposed
 issue.

Message for Open Source:
Fixed AR0231 overexposed issue.

Message for Internal:
[Issue/Feature]
Fixed AR0231 overexposed issue in HDR mode.

[Root Cause/Changes]
The source of this overexposed issue is due to
that LUT companding is disabled in HDR mode.

Change-Id: I836796b2f72b7e7c56823094cd84aeaea97a15a8
Signed-off-by: Chen, Yongsheng <yongsheng.chen@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h b/drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h
index d3107c6..9995f19 100644
--- a/drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h
+++ b/drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h
@@ -694,6 +694,7 @@ static struct crl_register_write_rep ar0231at_1920_1088_2hdr_mode[] = {
 	{ 0x3064, CRL_REG_LEN_16BIT, 0x1802, 0x10 },
 	{ 0x3064, CRL_REG_LEN_16BIT, 0x1802, 0x10 },
 	{ 0x3064, CRL_REG_LEN_16BIT, 0x1802, 0x10 },
+	{ 0x33DA, CRL_REG_LEN_16BIT, 0x0002, 0x10 },
 	{ 0x33E0, CRL_REG_LEN_16BIT, 0x0C80, 0x10 },
 	{ 0x33E0, CRL_REG_LEN_16BIT, 0x0C80, 0x10 },
 	{ 0x3180, CRL_REG_LEN_16BIT, 0x0080, 0x10 },
@@ -1033,6 +1034,7 @@ static struct crl_register_write_rep ar0231at_1920_1088_3hdr_mode[] = {
 	{ 0x3064, CRL_REG_LEN_16BIT, 0x1802, 0x10 },
 	{ 0x3064, CRL_REG_LEN_16BIT, 0x1802, 0x10 },
 	{ 0x3064, CRL_REG_LEN_16BIT, 0x1802, 0x10 },
+	{ 0x33DA, CRL_REG_LEN_16BIT, 0x0002, 0x10 },
 	{ 0x33E0, CRL_REG_LEN_16BIT, 0x0C80, 0x10 },
 	{ 0x33E0, CRL_REG_LEN_16BIT, 0x0C80, 0x10 },
 	{ 0x3180, CRL_REG_LEN_16BIT, 0x0080, 0x10 },
@@ -1372,6 +1374,7 @@ static struct crl_register_write_rep ar0231at_1920_1088_4hdr_mode[] = {
 	{ 0x3064, CRL_REG_LEN_16BIT, 0x1802, 0x10 },
 	{ 0x3064, CRL_REG_LEN_16BIT, 0x1802, 0x10 },
 	{ 0x3064, CRL_REG_LEN_16BIT, 0x1802, 0x10 },
+	{ 0x33DA, CRL_REG_LEN_16BIT, 0x0002, 0x10 },
 	{ 0x33E0, CRL_REG_LEN_16BIT, 0x0C80, 0x10 },
 	{ 0x33E0, CRL_REG_LEN_16BIT, 0x0C80, 0x10 },
 	{ 0x3180, CRL_REG_LEN_16BIT, 0x0080, 0x10 },
-- 
2.7.4

